include/crocoddyl/core/activations/quadratic-barrier.hpp

```diff
@@ -105,9 +107,14 @@ class ActivationModelQuadraticBarrierTpl : public ActivationModelAbstractTpl<_Sc
 
     boost::shared_ptr<Data> d = boost::static_pointer_cast<Data>(data);
     data->Ar = (d->rlb_min_ + d->rub_max_).matrix();
-    data->Arr.diagonal() = (((r - bounds_.lb).array() <= Scalar(0.)) + ((r - bounds_.ub).array() >= Scalar(0.)))
-                               .matrix()
-                               .template cast<Scalar>();
+
+    using pinocchio::internal::ComparisonOperators;
+    using pinocchio::internal::if_then_else;
+    for (Eigen::Index i = 0; i < data->Arr.cols(); i++) {
```

Comment:
More efficient to use ++i.

Reply:
There is no difference in modern compilers.